autoextract_poet/util.py

```diff
@@ -8,15 +8,35 @@
 CLASS_ATTRS: WeakKeyDictionary = WeakKeyDictionary()
 
 
-def remove_unknown_fields(data: Optional[dict], item_cls: Type) -> dict:
-    """Return a dict where those elements not belonging to the
-    attr class ``item_cls`` definition are removed. This dict is then safe to
-    be used to create items for the given class. Can be used to ensure
-    that new attributes returned from the API does not break the library"""
+def split_in_unknown_and_known_fields(data: Optional[dict], item_cls: Type) -> Tuple[Dict, Dict]:
+    """
+    Return a pair of dicts. The first one contains those elements not belonging to the
+    attr class ``item_cls``. The second one contains the rest. That is, those
+    attributes not belonging to ``item_cls`` class
+    """
     data = data or {}
     if not attr.has(item_cls):
         raise ValueError(f"The cls {item_cls} is not attr.s class")
-
     if not item_cls in CLASS_ATTRS:
         CLASS_ATTRS[item_cls] = {field.name for field in attr.fields(item_cls)}
-    return {k: v for k, v in data.items() if k in CLASS_ATTRS[item_cls]}
\ No newline at end of file
+    unknown, known = split_dict(data, lambda k: k in CLASS_ATTRS[item_cls])
```

Comment:
Do I understand correctly, that `unknown` would be filled only if we expect `data` to have attributes of, for example, `Rating` item class, but also it includes some additional properties? If so, how would these additional properties appear there (in `data`)?

Reply:
Not sure to understand the question. Let me explain a bit what is happening here. `data` is a dictionary with all the arguments to initialize the class. It might be a nested structure of dicts and lists, **but never containing other items**. In other words, it contains the data as it comes from the API. 

Then, we split the first level of attributes into two classes: known and unknown attributes. The known ones are those that are defined in the item (e.g. `name` for `Product`). The unknown ones are those not recognized. Those will go into the `_additional_attrs`.

Note that this mechanism is recursive. So, if some unknown attribute is detected in the dictionary corresponding to the `Rating`, it will be stored in `Rating._additional_attrs` and not in `Product._additional_attrs`

Hope this answer your question.